// Crear un programa en C++ que muestre como una funci칩n
// separada de la funci칩n principal puede
// enviar excepciones que ser치n capturadas en la funci칩n main,

#include <iostream>
using namespace std;
void fun(int *ptr, int x)
{
    if (ptr == NULL)
        throw ptr;
    if (x == 0)
        throw x;
    /* Alguna funcionalidad */
}
int main()
{
    try
    {
        fun(NULL, 0);
    }
    catch (...) // por defecto
    {
        cout << "Capturando la excepcion desde fun()" << endl;
    }
    return 0;
}